apps/emqx/src/emqx_channel.erl

```diff
@@ -842,6 +808,24 @@ do_subscribe(
             {RC, Channel}
     end.
 
+%% TODO: rm me
+gen_reason_codes(CheckedTopicFilters, TopicFiltersWithRC) ->
+    TupleTopicFilters2 =
+        lists:foldl(
+            %% previous `deny_subscription` in subopts never used in EMQX 5.x hooks.
+            %% And after `client.subscribe` hook modified, we can deny subscription by hooks
+            %% So we rm `deny_subscription` here.
+            fun(NTopicFilterWithRC = {Key, _Value}, Acc) ->
+                lists:keyreplace(Key, 1, Acc, NTopicFilterWithRC)
+            end,
+            CheckedTopicFilters,
+            TopicFiltersWithRC
+        ),
+    [
+        ReasonCode
+     || {_TopicFilter = {_, _}, ReasonCode} <- TupleTopicFilters2
+    ].
+
```

Comment:
rm this func after sub-hook changed or rm comment after we decide to keep previous behaviour.